extensions/msal-node-extensions/src/persistence/BasePersistence.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { PersistenceError } from "../error/PersistenceError";
+import { Constants } from "../utils/Constants";
+import { IPersistence } from "./IPersistence";
+
+export abstract class BasePersistence {
+    public abstract createForPersistenceValidation(): Promise<IPersistence>; 
+
+    public async verifyPersistence(): Promise<boolean> {
+        // We are using a different location for the test to avoid overriding the functional cache
+        const persistenceValidator = await this.createForPersistenceValidation();
+
+        try {
+            persistenceValidator.save(Constants.PERSISTENCE_TEST_DATA);
+
+            const retrievedDummyData = await persistenceValidator.load();
+
+            if (!retrievedDummyData) {
+                throw PersistenceError.createCachePersistenceError(
```

Comment:
Is it possible in other platforms? And can we also compare this with a design that can throw `platform` specific errors instead?

Reply:
This cases are derived from the dotnet node-extensions, from my own tests, I have not been able to derive more cases than covered here.

Reply:
Well, approving for now. Let us work on a better solution eventually to segregate platform errors.